src/Parser/Monk/Brewmaster/Modules/Core/MasteryValue.js

```diff
@@ -0,0 +1,298 @@
+ * number of stacks you can reach, since after that point every hit will
+ * be a dodge.
+ *
+ * The expected number of stacks you have is just the sum of the indices
+ * weighted by the values.
+ */
+class StackMarkovChain {
+  _stackProbs = [1.0];
+
+  _assertSum() {
+    const sum = this._stackProbs.reduce((sum, p) => p + sum, 0);
+    if (Math.abs(sum - 1.0) > 1e-6) {
+      const err = new Error("probabilities do not sum to 1 in StackMarkovChain");
+      err.data = {
+        sum: sum,
+        probs: this._stackProbs,
+      };
+      console.log(err.data);
+
+      throw err;
+    }
+  }
+
+  // add a stack with guaranteed probability
+  guaranteeStack() {
+    this._stackProbs.unshift(0.0);
+    this._assertSum();
+  }
+
+  processAttack(dodgeProb, masteryValue) {
+    this._stackProbs.push(0);
+    const n = this._stackProbs.length - 1;
+
+    // probability of ending at 0 stacks. initial 
+    let zeroProb = 0;
+    // didn't dodge, gain a stack
+    for(let stacks = n-1; stacks >= 0; stacks--) {
+      const prob = _clampProb(dodgeProb + masteryValue * stacks);
+      zeroProb += prob * this._stackProbs[stacks]; // dodge -> go to 0
+      const hitProb = 1 - prob;
+      this._stackProbs[stacks+1] = hitProb * this._stackProbs[stacks]; // hit -> go to stacks + 1
+    }
+    // did dodge, reset stacks
+    this._stackProbs[0] = zeroProb;
+    this._assertSum();
+  }
+
+  get expected() {
+    return this._stackProbs.reduce((sum, prob, index) => sum + prob * index, 0);
+  }
+}
+
+/**
+ * Estimate the expected value of mastery on this fight. The *actual*
+ * estimated value is subject to greater variance.
+ * On the other hand, the expected value averages over all
+ * possible outcomes and gives a better sense of how valuable mastery is
+ * if you were to do this fight again. This values more stable over
+ * repeated attempts or kills and reflects the value of mastery (and the
+ * execution of the rotation) more closely than how well-favored you
+ * were on this particular attempt.
+ *
+ * We calculate the expected value by applying the Markov Chain above to
+ * the timeline to calculate the expected number of stacks at each
+ * dodgeable event. This is then combined with information about the
+ * expected damage of each dodged hit (dodge events have amount: 0,
+ * absorbed: 0, etc.) to provide a best-guess estimate of the damage
+ * you'll mitigate on average. The actual estimate (shown in the
+ * tooltip) may be over or under this, but is unlikely to be far from
+ * it.
+ *
+ * This madness was authored by emallson. If you need further explanation of
+ * the theory behind it, find me on discord.
+ */
+class MasteryValue extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    dmg: DamageTaken,
+    stats: StatTracker,
+  };
+
+  _dodgeableSpells = {};
+  _timeline = [];
+  _hitCounts = {};
+  _dodgeCounts = {};
+
+  get baseDodge() {
+    return 0.15;
```

Comment:
Not clear on why this is in a getter instead of as a static const.

Reply:
I was hoping the other brm theorycrafters would have the formula for it (its a function of agility). Turns out nobody bothered computing it for legion because it hasn't really mattered. For right now I'm going to change it to a const and then when I get the formula i'll revisit it.